### PR TITLE
[tmva][pymva] Optimize Evaluation in PyKeras

### PR DESCRIPTION
**Pull request by lmoneta**
Disable tensorflow eager execution when evaluating the model in MethodPyKeras This speeds up by more than 100 the model prediction.

Re-arrange the code and fixes also a memory leak in allocating the input array

### CODE REVIEW

File: tmva/pymva/src/MethodPyKeras.cxx [hunk 1]
```diff
@@ -363,19 +363,36 @@ void MethodPyKeras::SetupKerasModel(bool loadTrainedModel) {
...
+}
+
+///Setting up model for evaluation
+/// Add here some needed optimizations like disabling eager execution
+void MethodPyKeras::SetupKerasModelForEval() {
+
+   // disable eager execution (model will evaluate > 100 faster)
+   // need to be done before loading the model
+   if (fUseTFKeras){
+      PyRunString("tf.compat.v1.disable_eager_execution()","Failed to disable eager execution");
```
Comment (omazapa) [inline tmva/pymva/src/MethodPyKeras.cxx:378]: very nice catch!

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, with flags -Dtmva-sofie=On 
-DCTEST_TEST_EXCLUDE_NONE=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default with flags -Dtmva-sofie=On -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default with flags -Dtmva-sofie=On -DCTEST_TEST_EXCLUDE_NONE=On

**Review by sanjibansg — APPROVED** [review]
This is a nice addition to the existing code! Optimizing the eager execution mode will speed-up the evaluation process.

**Pull request merged**